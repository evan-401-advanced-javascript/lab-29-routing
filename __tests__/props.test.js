import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import Message from '../src/components/message';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});


describe('up and down', () => {
  it('should show hello world', () => {
    const app = mount(<Message />);
    expect(app.state().text).toBe('hello world');
  });

 
  it('count should show a higher number in the DOM', () => {
    act(() => {
      ReactDOM.render(<Message />, container);
    });
    const label = container.querySelector('span');
    expect(label.textContent).toBe('0');
  });

  it('should show that the snapshot matches', () => {
    const tree = renderer.create(<Message />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
