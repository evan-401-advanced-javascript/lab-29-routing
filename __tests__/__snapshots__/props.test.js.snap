// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`up and down count should show a higher number in the DOM 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    id="down"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    id="up"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;

exports[`up and down should show that the snapshot matches 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    id="down"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    id="up"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
